src/main/java/xyz/nucleoid/plasmid/event/GameEvents.java

```diff
@@ -0,0 +1,72 @@
+package xyz.nucleoid.plasmid.event;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+import net.minecraft.util.Identifier;
+import xyz.nucleoid.plasmid.game.ConfiguredGame;
+import xyz.nucleoid.plasmid.game.GameCloseReason;
+import xyz.nucleoid.plasmid.game.GameSpace;
+import xyz.nucleoid.plasmid.game.StartResult;
+
+/**
+ * Events for games being opened and closed/finishing
+ */
+public final class GameEvents {
+
+    private GameEvents() { }
+
+    public static final Event<GameOpening> OPENING = EventFactory.createArrayBacked(GameOpening.class,
+            listeners -> (gameId, game, oneShot) -> {
+        for (GameOpening listener : listeners) {
+            listener.onGameOpening(gameId, game, oneShot);
+        }
+    });
+
+    public static final Event<GameOpen> OPEN = EventFactory.createArrayBacked(GameOpen.class,
+            listeners -> (gameId, game, oneShot, gameSpace) -> {
+        for (GameOpen listener : listeners) {
+            listener.onGameOpen(gameId, game, oneShot, gameSpace);
+        }
+    });
+
+    public static final Event<GameStartRequest> START_REQUEST = EventFactory.createArrayBacked(GameStartRequest.class,
+            listeners -> (gameSpace, result) -> {
+        for (GameStartRequest listener : listeners) {
+            listener.onRequestStart(gameSpace, result);
+        }
+    });
+
+    public static final Event<GameClosing> CLOSING = EventFactory.createArrayBacked(GameClosing.class,
+            listeners -> (gameSpace, reason) -> {
+        for (GameClosing listener : listeners) {
+            listener.onGameClosing(gameSpace, reason);
+        }
+    });
+
+    public interface GameOpening {
+        /**
+         * @param gameId The ID of the game config being opened
+         * @param game The game and its configuration
+         * @param oneShot If the game is a one off, or whether it is part of a channel
+         */
+        void onGameOpening(Identifier gameId, ConfiguredGame<?> game, boolean oneShot);
+    }
+
+    public interface GameOpen {
+        /**
+         * @param gameId THe ID of the game config that has been opened
+         * @param game The game and its configuration
+         * @param oneShot true if the game is a one off, false if it is part of a channel
+         * @param gameSpace The {@link GameSpace} the game is running in.
+         */
+        void onGameOpen(Identifier gameId, ConfiguredGame<?> game, boolean oneShot, GameSpace gameSpace);
```

Comment:
A bit conflicted whether it makes sense for this event to be called from the responsible channels as it is now, or rather whether it should be called from deeper inside the game opening logic itself which they both go through.
Also, is the `oneShot` parameter needed?

Reply:
I call the events from in the channels because it gives more access to context (game ID mostly) than inside `ConfiguredGame#open`. The `oneShot` parameter is there to give as much context as possible; for example, a mod may not what to do something when games are opened from a command

Reply:
Hmm- what about having like a `OneshotChannelOpen` event and then the regular game opening events which don't have access to game ID, but will always be called?

Generally relying on game config IDs is something to avoid, given they are not mandatory and it is valid to create an anonymous game config.

Reply:
Ah, I didn't fully realise it was possible to open games anonymously, I'll see if I can add a separate event for oneshots then